pallets/pallet-bonded-coins/src/curves/mod.rs

```diff
@@ -0,0 +1,228 @@
+///  Curve Module
+///
+/// This module defines various curve types and their associated parameters used
+/// in the system. It includes the following curve types:
+/// - Polynomial
+/// - SquareRoot
+/// - LMSR (Logarithmic Market Scoring Rule)
+pub(crate) mod lmsr;
+pub(crate) mod polynomial;
+pub(crate) mod square_root;
+
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_arithmetic::ArithmeticError;
+use sp_core::U256;
+use sp_runtime::traits::CheckedConversion;
+use sp_std::ops::{AddAssign, BitOrAssign, ShlAssign, ShrAssign};
+use substrate_fixed::traits::{Fixed, FixedSigned, ToFixed};
+
+use crate::{
+	curves::{
+		lmsr::{LMSRParameters, LMSRParametersInput},
+		polynomial::{PolynomialParameters, PolynomialParametersInput},
+		square_root::{SquareRootParameters, SquareRootParametersInput},
+	},
+	types::Round,
+	PassiveSupply, Precision,
+};
+
+/// An enum representing different types of curves with their respective
+/// parameters. Used to store curve parameters and perform calculations.
+#[derive(Clone, Debug, Encode, Decode, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub enum Curve<Parameter> {
+	Polynomial(PolynomialParameters<Parameter>),
+	SquareRoot(SquareRootParameters<Parameter>),
+	Lmsr(LMSRParameters<Parameter>),
+}
+
+/// An enum representing input parameters for different types of curves.
+/// Used to convert into Curve.
+#[derive(Clone, Debug, Encode, Decode, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub enum CurveInput<Parameter> {
```

Comment:
Coefficient indeed is a better name. And yes, currently the type is the same for all, but this could definitely change at some point. Just not sure it's urgent.

Reply:
[30019d1](https://github.com/KILTprotocol/kilt-node/pull/816/commits/30019d12d4bb1bdc243d5efc28d81e15a97a206c)